auth/src/main/java/com/firebase/ui/auth/util/SmartlockUtil.java

```diff
@@ -12,44 +12,54 @@
  * Helper class to deal with Smartlock Flows.
  */
 public class SmartlockUtil {
-
     /**
      * If SmartLock is enabled and Google Play Services is available, start the save credential
      * Activity. Otherwise, finish the calling Activity with RESULT_OK.
-     * @param activity the calling Activity.
-     * @param requestCode request code to use when starting the save operation.
-     * @param parameters calling Activity flow parameters.
+     *
+     * @param activity     the calling Activity.
+     * @param requestCode  request code to use when starting the save operation.
+     * @param data         An Intent to be merged. Must be a {@code new Intent().putExtras(...)}.
```

Comment:
Can you explain a little what happened when you did the merge the other way?  What's the buggy behavior you're seeing?